File: src/ros2_bridge/mappers/std_msgs/int32.rs
//! Typed mapper for `std_msgs/msg/Int32`.

use crate::ros2_bridge::mapper::TypedTopicMapper;

pub(crate) fn int32_to_bus(msg: ros_env::std_msgs::msg::Int32) -> crate::std_msgs::msg::v1::Int32 {
    crate::std_msgs::msg::v1::Int32 {
        data: msg.data,
    }
}

pub(crate) fn int32_to_ros(bus: crate::std_msgs::msg::v1::Int32) -> ros_env::std_msgs::msg::Int32 {
    ros_env::std_msgs::msg::Int32 {
        data: bus.data,
    }
}

#[derive(Clone, Copy, Debug, Default)]
pub struct StdMsgsInt32Mapper;

impl TypedTopicMapper for StdMsgsInt32Mapper {
    type Ros = ros_env::std_msgs::msg::Int32;
    type Bus = crate::std_msgs::msg::v1::Int32;

    fn type_name(&self) -> &'static str {
        "std_msgs/msg/Int32"
    }

    fn ros_to_bus(&self, msg: Self::Ros) -> crate::errors::Result<Self::Bus> {
        Ok(int32_to_bus(msg))
    }

    fn bus_to_ros(&self, msg: Self::Bus) -> crate::errors::Result<Self::Ros> {
        Ok(int32_to_ros(msg))
    }
}

File: bindings/python/robot_bus/ros2_bridge/mappers/std_msgs/int32.py
"""Generated mapper for `std_msgs/msg/Int32`."""

from __future__ import annotations

from robot_bus.ros2_bridge.mappers import _convert


def int32_to_bus(msg):
    from robot_bus.std_msgs.msg.v1 import Int32 as BusMsg

    bus = BusMsg()
    bus.data = msg.data
    return bus


def int32_to_ros(bus):
    from std_msgs.msg import Int32 as RosMsg

    out = RosMsg()
    out.data = bus.data
    return out


class StdMsgsInt32Mapper:
    def type_name(self) -> str:
        return "std_msgs/msg/Int32"

    def ros_msg_type(self):
        from std_msgs.msg import Int32 as RosMsg

        return RosMsg

    def ros_to_bus(self, msg) -> bytes:
        return int32_to_bus(msg).SerializeToString()

    def bus_to_ros(self, payload: bytes):
        from robot_bus.std_msgs.msg.v1 import Int32 as BusMsg

        bus = BusMsg()
        bus.ParseFromString(payload)
        return int32_to_ros(bus)
